command/agent/template/template.go

```diff
@@ -255,16 +255,35 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		ServerName: pointerutil.StringPtr(""),
 	}
 
-	if strings.HasPrefix(sc.VaultConf.Address, "https") || sc.VaultConf.CACert != "" {
-		skipVerify := sc.VaultConf.TLSSkipVerify
+	// Use the cache if available or fallback to the Vault server values.
+	if sc.AgentConfig.Cache != nil && len(sc.AgentConfig.Listeners) != 0 {
+		scheme := "unix://"
+		if sc.AgentConfig.Listeners[0].Type == "tcp" {
+			scheme = "https://"
+			if sc.AgentConfig.Listeners[0].TLSDisable {
+				scheme = "http://"
+			}
+		}
+		address := fmt.Sprintf("%s%s", scheme, sc.AgentConfig.Listeners[0].Address)
+		conf.Vault.Address = &address
+
+		// Only configure TLS Skip Verify if CT is not going through the cache. We can
+		// skip verification if its using the cache because they're part of the same agent.
+		if scheme == "https" {
+			conf.Vault.SSL.Verify = pointerutil.BoolPtr(false)
+			conf.Vault.SSL.Cert = pointerutil.StringPtr("")
+			conf.Vault.SSL.Key = pointerutil.StringPtr("")
```

Comment:
Don't we have to provide `conf.Vault.SSL.CaCert` if `sc.AgentConfig.Vault.Listeners[0].TLSCertFile` got provided? 

I haven't tested this manually, but it looks like mTLS [can be enabled/enforced on the listener](https://www.vaultproject.io/docs/configuration/listener/tcp#tls_client_ca_file) if they provide `tls_client_ca_file` and they don't toggle on `tls_require_and_verify_client_cert` so I'm not sure hardcoding `Cert` and `Key` to empty string is proper in those cases.

Reply:
@calvn `conf.Vault.SSL.Verify = pointerutil.BoolPtr(false)` disables verification, which is what the CA is used for.

Updated cert/key.